pkg/build/api/types.go

```diff
@@ -25,6 +25,8 @@ const (
 	BuildJenkinsLogURLAnnotation = "openshift.io/jenkins-log-url"
 	// BuildJenkinsBuildURIAnnotation is an annotation holding a link to the Jenkins build
 	BuildJenkinsBuildURIAnnotation = "openshift.io/jenkins-build-uri"
+	// BuildMatchSourceURIAnnotation is an annotation on a Secret indicating the URIs for which it can be used
+	BuildMatchSourceURIAnnotation = "build.openshift.io/match-source-uri"
```

Comment:
> Not to mention the UI will also need to look for both values in a lot of these cases.
> 
> I think we need a better justification for such a migration.

I wasn't suggesting this pull do a migration.  Do you see any of that as a reason to perpetuate a pattern which no longer serves us?  "consistency" just means, "forever be inconsistent with any new API group being developed", so it doesn't hold up over time.